ConsistencyManager/DataStructures/WeakArray.swift

```diff
@@ -117,14 +117,19 @@ public struct WeakArray<T: AnyObject>: ArrayLiteralConvertible {
 
 // MARK: MutableCollectionType Implementation
 
-extension WeakArray: MutableCollectionType {
+extension WeakArray: MutableCollection {
 
     // Required by SequenceType
-    public func generate() -> IndexingGenerator<WeakArray<T>> {
+    public func makeIterator() -> IndexingIterator<WeakArray<T>> {
         // Rather than implement our own generator, let's take advantage of the generator provided by IndexingGenerator
-        return IndexingGenerator<WeakArray<T>>(self)
+        return IndexingIterator<WeakArray<T>>(_elements: self)
     }
 
+    public func index(after i: Int) -> Int {
+        //TODO: modify the code to return the correct value
```

Comment:
This is a required method for the protocol implementation. I believe this is the right implementation. I wasn't completely sure so I just added comment.